lib/blog_helpers.rb

```diff
@@ -51,25 +55,29 @@ def related_articles(article, limit = 3)
 
   private
 
-  def sepcontent
+  def content
     data_directory = if @app.data[:sepcontent].nil? || ENV["DATA_DIR"].eql?("sample")
       :sample
     else
       :sepcontent
     end
-    @sepcontent ||= @app.data[data_directory]
+    @content ||= @app.data[data_directory]
   end
 
   def article_entries
-    if sepcontent && sepcontent.articles
-      structurize(sepcontent.articles)
+    @article_entries ||= if content && content.articles
+      structurize(content.articles)
         .sort_by(&:date)
         .reverse
     end
   end
 
+  def page_entries
+    @page_entries ||= content.pages.map { |page| OpenStruct.new(page[1]) } if content && content.pages
```

Comment:
Usually using OpenStruct includes huge performance penalties. Some more info
http://stackoverflow.com/questions/14421218/when-should-i-use-an-openstruct-instead-of-a-hash

there are also other issues with OpenStruct that are not mentioned, but are easily googlable.

Are you definitely certain that this can't be written differently?